File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    rank = [6,6,5,4,3,2,1]
    l_count = h_count = 0

    for i in lottos:
        if i != 0:
            if i in win_nums:
                l_count += 1
                h_count += 1
        else:
            h_count += 1
    answer.append(rank[h_count])
    answer.append(rank[l_count])
    return answer

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    const prize = {
        '6' : 1,
        '5' : 2,
        '4' : 3,
        '3' : 4,
        '2' : 5
    };
    let zero_count = 0;
    let correct = 0;
    
    for(val of lottos){
        if(win_nums.includes(val)) correct+=1;
        if(val === 0) zero_count+=1;
    };
    const max = correct+zero_count;
    const min = correct;
    return [(max < 2) ? 6 : prize[max], (min < 2) ? 6 : prize[min]];
}